src/git.js

```diff
@@ -0,0 +1,75 @@
+const gh = require('parse-github-url');
+const parsePath = require('parse-filepath');
+
```

Comment:
Employ your factory pattern here. Rather than commingling the github and gitlab logic in multiple if/else blocks, use the factory to produce a module for github that just knows how to handle github, and the same for gitlab. This will keep the different git repo implementation separate, and simplify adding the next one.